lib/dry/system/provider_registrar.rb

```diff
@@ -215,6 +228,7 @@ def build_provider_from_source(name, source:, group:, options:, &block)
           name: name,
           target_container: target_container,
           source_class: provider_source.source,
+          source_options: provider_source_options,
```

Comment:
I wonder whether we want to be more careful before passing `provider_source_options` here, because in this case, the provider registrar doesn't have the same control over the `source_class` as it does in `build_provider`, so we can't be 100% confident it will even accept any extra options.

Perhaps we want to do something like this?

```
source_options = provider_source.source == provider_source_class ? provider_source_options : {}
```

And then pass that in? This way we're only passing custom options if we know the source class matches what the registrar expects it to be.